File: Python_Solutions/insertion_sort/insertion_sort.py
# Implement an insertion sort algorithm

def insertion_sort(arr):
    for i in range(1, len(arr)):
        j = i - 1
        currentValue = arr[i]
        while j >= 0 and arr[j] > currentValue:
            arr[j+1] = arr[j]
            j -= 1
        arr[j+1] = currentValue
    return arr


print(insertion_sort([4, 1, 2, 51, 24, 56, 32, 12, 4,
      7, 100, 9, -1, -12, 21, 30, 34, -3, -21, 31, 34, -200, 300]))

File: Js_Solutions/insertion_sort/insertion_sort.js
// Implement an insertion sort algorithm

function insertionSort(arr) {
    for (let i = 1; i < arr.length; i++) {
        let j;
        let currentVal = arr[i];
        for (j = i - 1; j >= 0 && arr[j] > currentVal; j--) {
            arr[j + 1] = arr[j];
        }
        arr[j + 1] = currentVal
    }
    return arr;
}

console.log(insertionSort([4, 1, 2, 51, 24, 56, 32, 12, 4, 7, 100, 9, -1, -12, 21, 30, 34, -3, -21, 31, 34]));